lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -742,6 +759,29 @@ def generate_certificate(
             extension.value,
             critical=extension.critical,
         )
+
+    # Filter out default extensions that were overridden by the caller. Without this step we'd get
+    # ValueError: This extension has already been set.
+    additional_extensions = additional_extensions or []
+    additional_extensions_oids = [extension.oid for extension in additional_extensions]
+    default_extensions = [
+        extension
+        for extension in {
+            Extension(
+                oid=x509.ExtensionOID.BASIC_CONSTRAINTS,
+                value=x509.BasicConstraints(ca=False, path_length=None),
+                critical=False,
+            )
+        }
+        if extension.oid not in additional_extensions_oids
```

Comment:
I disagree with this way of filtering, particularly for the `BasicConstraints` extension, as that means that someone could create a CA certificate by providing that extension.

I would keep complete control over most extensions and only support adding Subject Alternative Name. I do not think we currently have a good use case to allow more than that, and it could really reduce security.

Reply:
> someone could create a CA certificate by providing that extension

Correct, and I imagine that's what they'd need to do to create an intermediate CA.

I am fine with dropping the `additional_extensions` arg, but without the newly added extensions the cert doesn't pass strict verification.

Reply:
I would definitely keep `AuthorityKeyIdentifier` and `SubjectKeyIdentifier`, as I think they are a good addition even if they should be optional according to the RFC. `BasicConstraints` I am less sure of. Are you saying that not providing that extension makes the validation fail? I would find it bizarre that a non-critical extension can fail the validation.

> > someone could create a CA certificate by providing that extension
> 
> Correct, and I imagine that's what they'd need to do to create an intermediate CA.

I just did an audit of the code by reading the RFC, and I plan on updating things further after this PR, including a limit on `path_length` in the CA certificate set at 0 by default to prevent this use case. Currently, our CA certificates will allow any number of intermediary CA, and I do not think this is a good security posture as a default.